ui/app/components/clients/attribution.js

```diff
@@ -103,9 +104,24 @@ export default class Attribution extends Component {
     }
   }
 
-  get getCsvData() {
+  destructureCountsToArray(object) {
+    let { clients, entity_clients, non_entity_clients } = object;
+    return [clients, entity_clients, non_entity_clients];
+  }
+
+  constructCsvRow(firstColumn, secondColumn = null, otherColumns) {
+    // ['ns label', 'mount label', 'total clients', 'entity', 'non-entity']
+    // if constructing namespace row, mount column will be blank
+    // if firstColumn is a string, then we're at mount level attribution
+    return [
+      `${typeof firstColumn === 'string' ? firstColumn : firstColumn.label}`,
+      `${secondColumn ? secondColumn.label : ''}`,
+      ...otherColumns,
+    ];
+  }
+  generateCsvData() {
+    let graphData = this.args.totalClientAttribution;
```

Comment:
what happens if totalClientAttribution is empty?  Does it return an empty array?  I ask because of the forEach on 134.

Reply:
higher up in this file there is this getter, so the export button only shows if there is data to iterate and export

  ```
  get hasCsvData() {
    return this.args.totalClientAttribution ? this.args.totalClientAttribution.length > 0 : false;
  }
  ```